src/scipp/core/bins.py

```diff
@@ -378,8 +379,37 @@ def _groupby_bins(obj):
     return GroupbyBins(obj)
 
 
-def histogram(x: Union[_cpp.DataArray, _cpp.Dataset], *,
-              bins: _cpp.Variable) -> Union[_cpp.DataArray, _cpp.Dataset]:
+def _get_coord(x, name):
+    event_coord = x.bins.meta.get(name) if x.bins is not None else None
+    return x.meta.get(name, event_coord)
+
+
+def _upper_bound(x):
+    import numpy as np
+    bound = x.max()
+    bound.value = np.nextafter(bound.value, np.inf)
+    return bound
+
+
+def _parse_coords_arg(x, name, arg):
+    coord = _get_coord(x, name)
+    if isinstance(arg, int):
+        coord = linspace(name, coord.min(), _upper_bound(coord), num=arg + 1)
+    elif isinstance(arg, Variable):
+        if arg.ndim == 0:
```

Comment:
Is this a relevant case in practice? When does one have a known bin size but arbitrary boundaries?

Also, What about a scalar variable with `unit = None`? Would this be like passing an `int`?

Reply:
I think it is relevant. Someone may not care about the bounds, but knows their instrument resolution and want "0.1 Angstrom" bins.

Good point, need to avoid calling `to_unit` when `unit = None`. No, it would *not* be like passing and `int`.